WordPress/gutenberg
# Move entity-provider.js exports into hooks/index.ts so they are added to the docs

## What?
The hooks defined in `core-data/entity-provider.js` are not documented because the file  is not being pulled into the README.md for docgen to generate.

Closes: #63475.

## Why?
These hooks should be documented under https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#hooks

## How?
I have move all of the hooks in `entity-provider.js` into their own files inside the `hooks` directory and exported them from there. This will have them be picked up by the docgen tool and add them to the documentation.

## Testing Instructions
- Checkout branch and build the new changes
- Confirm that `useEntityProp` is still available on the `wp.coreData` global and there are no errors in the block editor.
